draft/celery-kubernetes-operator.rst

```diff
@@ -0,0 +1,273 @@
+.. vale off
+
+============================================================
+CEP XXXX: Celery Kubernetes Operator - Architecture Document
+============================================================
+
+:CEP: XXXX
+:Author: Gautam Prajapati
+:Implementation Team: Gautam Prajapati
+:Shepherd: Omer Katz, Asif Saif Uddin
+:Status: Draft
+:Type: Feature
+:Created: 2020-08-30
+:Last-Modified: 2020-08-30
+
+.. contents:: Table of Contents
+   :depth: 3
+   :local:
+
+Abstract
+========
+
+Kubernetes as a container orchestrator is nowadays a popular choice for deploying applications. To run Celery in production on Kubernetes, there are manual steps involved like -
+
+	* Writing deployment spec for workers
+	* Setting up monitoring using `Flower <https://flower.readthedocs.io/en/latest/>`_ 
+	* Setting up Autoscaling
+
+Apart from that, there’s no standard or consistent way to set up multiple clusters, everyone configures their own way which could create problems for infrastructure teams to manage and audit later.
+
+This project attempts to solve(or automate) these issues. It is aiming to bridge the gap between application engineers and infrastructure operators who manually manage the celery clusters.
+
+Moreover, since Celery is written in Python, we plan to use open-source `KOPF <https://github.com/nolar/kopf>`_\ (Kubernetes Operator Pythonic Framework) to write the custom controller implementation. 
+
+This CEP is currently written keeping Celery 4.X in mind. Controller implementation will differ for Celery 5 and is under discussion.
+
+
+Scope
+=====
```

Comment:
Style nitpick:
Our template includes all primary headings for every CEP.
If you have other headings, they should be included in one of the sections defined by the template.